Source/KeyHandler.mm

```diff
@@ -266,7 +266,7 @@ - (BOOL)handleInput:(KeyHandlerInput *)input state:(InputState *)inState stateCa
         return YES;
     }
 
-    if ([input isNumericPad]) {
+    if (!Preferences.selectCandidateWithNumericKeypad && [input isNumericPad]) {
```

Comment:
How about switching the evaluation order here? That is, change the predicate to: `[input isNumericPad] && !Preferences.selectCandidateWithNumericKeypad`, so that `Preferences` is only checked when really needed, whereas right now it's always checked even if the key event is not relevant.

Reply:
https://github.com/openvanilla/McBopomofo/pull/381/commits/ec8d9ff1bea2cdc1f072fff7e209d35ea9548a16